src/server/data_store/lru_cache_data_store.cc

```diff
@@ -0,0 +1,51 @@
+#include "lru_cache_data_store.h"
+
+::std::string KeyValuePair::get_key() const {
+  return key;
+}
+
+::std::string KeyValuePair::get_value() const {
+  return value;
+}
+
+// This is not thread safe and very naive implementation.
+::LruCacheDataStore::LruCacheDataStore(::std::uint32_t capacity) {
+  capacity = capacity;
+  key_iterator_map = ::std::unordered_map<::std::string, ::std::list<KeyValuePair>::iterator>();
+  pairs = ::std::list<KeyValuePair>();
+}
+
+::LruCacheDataStore::~LruCacheDataStore() {}
+
+::std::string LruCacheDataStore::Get(::std::string key) {
+  if (key_iterator_map.find(key) == key_iterator_map.end()) {
+    return nullptr;
+  }
+  ::std::list<KeyValuePair>::const_iterator existing_pair_iterator = key_iterator_map.at(key);
+  // TODO(joejoevictor): This seems dumb. I wonder if we could avoid the copy.
```

Comment:
I agree with you that this seems a little bit messy. I feel like we should abstract out the "LRU" part of the business logic into either helper functions/class and manage that, because some part of it will be shared between both set/get method. 

I am totally fine if we want to refactor it later.

Reply:
Not sure what the best way is to refactor the stuff here. There's basically 2 parts: removing from both list and map, inserting to both list and map.

The messiness is coming from the fact that sometimes we need to remove an element in the middle of the list, other times the head of the list. Not sure if there's too much we could put into helper function at this stage.

What do you think?